### PR TITLE
Fix TT-520 provide introspection for proxy mode

### PR DESCRIPTION
**Pull request by spetrunin**
## Description

Extract reverseProxy sendRequestToUpstream to separate method
Handle introspection for proxy mode

## Related Issue

https://tyktech.atlassian.net/browse/TT-520

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [ ] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
  - [ ] If new config option added, ensure that it can be set via ENV variable
- [ ] I have updated the documentation accordingly.
- [x] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] When updating library version must provide reason/explanation for this update.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] Check your code additions will not fail linting checks:
  - [x] `go fmt -s`
  - [x] `go vet`

### CODE REVIEW

File: gateway/reverse_proxy.go [hunk 1]
```diff
@@ -695,6 +695,74 @@ func (rt *TykRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
...
+		return
+	}
+
+	var buf bytes.Buffer
+	err = p.TykAPISpec.GraphQLExecutor.Schema.IntrospectionResponse(&buf)
+	if err != nil {
+		return
+	}
+
+	res = &http.Response{
```
Comment (furkansenharputlu) [inline gateway/reverse_proxy.go:756]: I think this part should be done on library side as we did above.

**Review by furkansenharputlu — APPROVED** [review]

**Review by pvormste — APPROVED** [review]

**Comment by pvormste** [comment]
manually merged to `release-3.1.0`

**Comment by pvormste** [comment]
manually merged to `release-3.1`

**Pull request merged**